models/accm.py

```diff
@@ -38,12 +38,13 @@ def __init__(self, input_layer, h_channels, number_of_rsacm):
             [(RsacmBlock(input_layer, self.h_channels)) for i in range(0, self.number_of_rsacm)])
 
     def forward(self, x):
+        x = x.detach() # Detach x make sure that gradient for this layer won't flow to previous layers.
         self.x_list = []
         x_copy = x.clone()
         x = self.rsacm_list[0](x_copy)
         self.x_list.append(x)

```

Comment:
If possible, it might be a good idea to avoid state, which often generates bugs when code is developed over long time horizons. I usually write a separate function that performs the forward computation and along with the result returns the necessary intermediate results - explicitly to avoid saving state in the model. It may not be possible here, though.